pkg/shim/shim.go

```diff
@@ -343,15 +349,27 @@ func (e *exe) parameters(uid, gid int) []string {
 func (e *exe) prepare(ctx context.Context, home string, fd, uid, gid int) *exec.Cmd {
 	params := e.parameters(uid, gid)
 	cmd := exec.CommandContext(ctx, params[0], params[1:]...)
-	cmd.Stdout = nil // nsenter and unshare do not log
-	cmd.Stderr = nil // nsenter and unshare do not log
+
+	outfd, err := os.OpenFile(e.env.OutPipe, os.O_WRONLY, 0700)
+	if err != nil {
+		panic(err)
+	}
+	cmd.Stdout = outfd
+
+	errfd, err := os.OpenFile(e.env.OutPipe, os.O_WRONLY, 0700)
+	if err != nil {
+		panic(err)
+	}
+	cmd.Stderr = errfd // nsenter and unshare do not log
```

Comment:
Obviously the panics have to go, and I need to store the FDs somewhere they can be closed anyway.